docs/DeveloperGuide.md

```diff
@@ -1,38 +1,135 @@
-# Developer Guide
+# Developer Guide for SpendSwift
 
 ## Acknowledgements
-
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
-
-## Design & implementation
-
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
-
-
-## Product scope
-### Target user profile
-
-{Describe the target user profile}
-
-### Value proposition
-
-{Describe the value proposition: what problem does it solve?}
+{List here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well.}
+
+## Design & Implementation
+### Sequence Diagrams
+#### add-expense
+![Add Expense Sequence Diagram](diagrams/AddExpense.png)
+
+#### add-category
+![Add Category Sequence Diagram](diagrams/AddCategory.png)
+
+#### delete-expense
+![Delete Expense Sequence Diagram](diagrams/DeleteExpense.png)
+
+#### tag-expense
+![Tag Expense Sequence Diagram](diagrams/TagExpense.png)
+
+#### set-budget
+![Set Budget Sequence Diagram](diagrams/SetBudget.png)
+
+#### view-budget
+![View Budget Sequence Diagram](diagrams/ViewBudget.png)
+
+#### view-expenses
+![View Expenses Sequence Diagram](diagrams/ViewExpenses.png)
+
+#### toggle-reset
+![Toggle Auto Reset Sequence Diagram](diagrams/ToggleAutoReset.png)
+
+### Class Diagrams
+#### Budget class
+![Budget Class Diagram](diagrams/budgetclass.png)
+
+#### Expense class
+![Expense Class Diagram](diagrams/expenseclass.png)
+
+#### Category class
+![Category Class Diagram](diagrams/CategoryClass.png)
+
+#### ExpenseTracker class
+![ExpenseTracker Class Diagram](diagrams/ExpenseTrackerclass.png)
+
+#### Duke class
```

Comment:
![image](https://github.com/user-attachments/assets/466a26d6-68f6-49ff-ad54-1da98b0246f9)

You should use the open arrow here.